.circleci/config.yml

```
@@ -0,0 +1,124 @@
+version: 2
+jobs:
+  build:
+    working_directory: ~/learninglocker
+    machine:
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run:
+          name: create cache directory
+          command: mkdir -p docker/images
+
+      - restore_cache:
+          keys:
+            - learninglocker-image-{{ .Revision }}
+            - learninglocker-image
+
+      - run:
+          name: Load images to docker engine
+          command: |
+            set -x
+            if [[ -e docker/images/learninglocker.tar ]]; then
+              docker load < docker/images/learninglocker.tar
+            fi
+
+      - run:
+          name: register build arg instruction to use it in build step if CIRCLE_TAG is not null
+          command: |
+            set -x
+            if [[ -n ${CIRCLE_TAG:+x} ]]; then
+              echo 'export ARG_VERSION="--build-arg LL_VERSION=${CIRCLE_TAG}"' >> $BASH_ENV
+            else
+              echo 'export ARG_VERSION=""' >> $BASH_ENV
+            fi
+            source $BASH_ENV
+
+      - run:
+          name: build learning locker image
+          command: |
+            docker build \
+              -t fundocker/learninglocker:${CIRCLE_SHA1} \
+              ${ARG_VERSION} \
```

Have you tried the following to remove the previous step?

```suggestion
              --build-arg LL_VERSION=${CIRCLE_TAG:-v2.6.2}" \
```